### PR TITLE
empty dataset

### PR DESCRIPTION
**Pull request by tomsilver**

### CODE REVIEW

File: src/approaches/interactive_learning_approach.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ def _get_current_predicates(self) -> Set[Predicate]:
     ######################## Semi-supervised learning #########################
 
     def learn_from_offline_dataset(self, dataset: Dataset) -> None:
+        # Special case: empty offline dataset. Has no annotations.
+        if not dataset.trajectories:
+            return
```
Comment (ronuchit) [inline src/approaches/interactive_learning_approach.py:43]: what breaks if you don't do this?

File: tests/datasets/test_datasets.py [hunk 2]
```diff
@@ -221,3 +221,15 @@ def test_ground_atom_dataset():
...
+def test_empty_dataset():
+    """Test creation of empty dataset."""
+    utils.reset_config({
+        "env": "cover",
+        "offline_data_method": "empty",
+    })
+    env = CoverEnv()
+    train_tasks = env.get_train_tasks()
+    dataset = create_dataset(env, train_tasks)
+    assert len(dataset.trajectories) == 0
```
Comment (ronuchit) [inline tests/datasets/test_datasets.py:235]: also assert something about annotations?

File: src/approaches/interactive_learning_approach.py (hunk 1 above)
Comment (ronuchit) [reply src/approaches/interactive_learning_approach.py:43]: oh, is it that dataset.annotations will be None?

**Review by ronuchit — APPROVED** [review]

**Review by amburger66 — APPROVED** [review]

File: src/approaches/interactive_learning_approach.py (hunk 1 above)
Comment (tomsilver) [reply src/approaches/interactive_learning_approach.py:43]: yeah (see the comment)

Comment (ronuchit) [reply src/approaches/interactive_learning_approach.py:43]: oh i see, it's not None it'll actually just crash upon access. nvm

**Pull request merged**